pydantic_ai_slim/pydantic_ai/models/cohere.py

```diff
@@ -242,7 +242,10 @@ def _map_user_message(cls, message: ModelRequest) -> Iterable[ChatMessageV2]:
             if isinstance(part, SystemPromptPart):
                 yield SystemChatMessageV2(role='system', content=part.content)
             elif isinstance(part, UserPromptPart):
-                yield UserChatMessageV2(role='user', content=part.content)
+                if isinstance(part.content, str):
+                    yield UserChatMessageV2(role='user', content=part.content)
+                else:
+                    raise ValueError('Only string as user prompt is supported for Cohere.')
```

Comment:
```suggestion
                    raise NotImplementedError('Only string as user prompt is supported for Cohere.')
```
or
```suggestion
                    raise RuntimeError('Only string as user prompt is supported for Cohere.')
```
If Cohere doesn't support multimodal, I'd use RuntimeError because this could get hit via fallback model or just using cohere instead of a model that _does_ support multi-modal inputs. But it's not a big deal.

If Cohere _does_ support multimodal and we just haven't implemented it yet, I'd do NotImplementedError.

Up to you if you want to handle it differently/as you've done it for now.